src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
+
+  let sum = 0;
+
+  women.forEach((w) => sum += w.died - w.born);
+
+  return sum / women.length;
```

Comment:
```suggestion
  return women.reduce((sumOfAges, woman) => (
    sumOfAges + woman.died - woman.born, 0)) / women.length
```

Avoid using loop and forEach